packages/components/atoms/f-button/src/components/Button.vue

```diff
@@ -105,72 +107,101 @@ export default {
         getAriaLive () {
             return this.isLoading ? 'polite' : 'off';
         }
+    },
+    watch: {
+        $props: {
+            immediate: true,
+            handler () {
+                this.validateProps();
+            }
+        }
```

Comment:
It's frustrating that this is the way mutually dependent prop validation has to be done when prop validators are supposed to be a thing, but [this](https://stackoverflow.com/questions/56904327/access-props-value-in-another-props-validator) would seem to indicate it is the accepted way. 

As a slightly cleaner solution, could I suggest that the list of valid button types for each component type not intersect, and therefore the 'isIcon' prop would become unnecessary, as the component type could be inferred from the button type? E.g.
```
export const VALID_BUTTON_TYPES = {
        iconButton: ['primaryIcon', 'secondaryIcon', 'ghostIcon', 'ghostTertiaryIcon'],
        button: ['primary', 'secondary', 'outline', 'ghost', 'link']
};
```
Or would that complicate things elsewhere?

Reply:
That could potentially work too, for sure. I'll give it a try when I get chance and see if it's an easy thing to switch to.

It would be a major bump (as the contract with the component would change for consumers potentially), but other than that I can't see any issues with it.

Reply:
I Have discussed this with @benwhite-deltas – after some investigation, it's not possible to get rid of the `isIcon` prop without also losing the prop validation, so will keep this like it is for now.